cartridge-cli.lua

```diff
@@ -2670,34 +2873,48 @@ local function app_pack(args)
         end
     end
 
-    if args.type == 'rpm' then
-        pack_rpm({
-            unit_template = unit_template,
-            instantiated_unit_template = instantiated_unit_template
-        })
-    elseif args.type == 'deb' then
-        pack_deb({
+    local opts
+    if array_contains({distribution_types.RPM, distribution_types.DEB}, args.type) then
+        opts = {
             unit_template = unit_template,
             instantiated_unit_template = instantiated_unit_template
-        })
-    elseif args.type == 'tgz' then
-        pack_tgz()
-    elseif args.type == 'rock' then
-        pack_rock()
-    elseif args.type == 'docker' then
-        pack_docker({
+        }
+    elseif args.type == distribution_types.DOCKER then
+        opts = {
             tag = args.tag,
             from = args.from,
             download_token = args.download_token,
             docker_build_args = args.docker_build_args,
-        })
-    else
-        die("Unknown package type: %s", args.type)
+        }
+    end
+
+    local pack_handler = pack_handlers[args.type]
+    if pack_handler == nil then
+        local handler_err = string.format("Pack handler for %s distribution type not found", args.type)
+        die(
+            "Whoops! It looks like something is wrong with this version of Cartridge CLI. " ..
+            "Please, report a bug on https://github.com/tarantool/cartridge-cli/issues/new. " ..
+            "The error is: %s.", handler_err
+        )
+    end
+
+    local ok_pack, err_pack = pack_handler(opts)
+    if not ok_pack then
+        warn('Failed to pack application')
```

Comment:
Where is an error message here (err_pack)?

Reply:
The main idea of this warning was to explain user why his build process suddenly started to clean build directory...
I will think about it and make log more user-friendly and informative.

Reply:
I mean that it should be `warn('Failed to pack application: %s', err_pack)`

Reply:
I thought that it should be smth like:
```
Packing rpm file
WARNING: Failed to pack application
Remove build directory ~/.cartridge/tmp/cartridge-build-fc3c12ba73e6b743
ERROR: Failed to pack application: Some error occured
```

Reply:
```
WARNING: Failed to pack application
ERROR: Failed to pack application: Some error occured
```

It's a duplication

Reply:
Do you suggest smth like this?

```
ERROR: Failed to pack application: Some error occured
Remove build directory...
```

Reply:
Yes

Reply:
The main problem is that `die` function writes message with `ERROR` prefix and then calls `os.exit(1)`. Your suggestion requires adding `err` function that writes `ERROR` message and doesn't call `os.exit(1)` (to allow performing cleanup). Then, `os.exit` should be called from `app_pack`.

I don't want to separate these actions, I suggest using:

```
WARNING: Failed to pack application                                 # Notify user that smth wrong
Remove build directory...                                                      # Cleanup
ERROR: Failed to pack application: Some error occured   # Explain the error
```